Assets/Scripts/GameAI/GameObjects/AIGameObject.cs

```diff
@@ -0,0 +1,241 @@
+﻿namespace GameAI.AIGameObjects
+{
+    using Navigation;
+    using GamePhysics;
+    using UnityEngine;
+    using StateHandlers;
+
+    public abstract class AIGameObject : MonoBehaviour
```

Comment:
The scope of this class seems a little big, but is okay for now.

Reply:
My goal was to have a one stop destination for enemy editor variables so that we only need to attach one script to an enemy for AI stuff. Which does have the downside of handling a lot of things in one class. I don't think it's in a place where it needs to be split up yet, but I might refactor it later down the line when more stuff gets implemented.